### ISSUE TITLE
Crash on installation: error: no matching function for call to ‘v8::FunctionTemplate::GetFunction()’

### ISSUE THREAD

**Issue by sbrl** [gpujs/gpu.js#468]
![A GIF or MEME to give some spice of the internet](url)

## *What* is wrong?
I can't install gpu.js on Ubuntu 18.10.

## *Where* does it happen?
In the terminal, if I type this: `npm install gpu.js`

## *How* do we replicate the issue?
On an Ubuntu 18.10 machine, do `npm install gpu.js`.

## *How* important is this (1-5)?
I'd assume it's a pretty game-breaking issue. 4-5.

## Expected behavior (i.e. solution)
I should be able to install it

## Other Comments
Output:

```
  CXX(target) Release/obj.target/translator_static/angle/src/compiler/translator/ShaderVars.o
  AR(target) Release/obj.target/angle/src/translator_static.a
  COPY Release/translator_static.a
  CXX(target) Release/obj.target/webgl/src/bindings.o
In file included from ../src/webgl.h:10,
                 from ../src/bindings.cc:9:
../../nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../../nan/nan.h:2298:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
../src/bindings.cc: In function ‘void Init(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’:
../src/bindings.cc:482:35: error: no matching function for call to ‘v8::FunctionTemplate::GetFunction()’
     , webgl_template->GetFunction());
                                   ^
In file included from /home/sbrl/.node-gyp/12.2.0/include/node/node.h:63,
                 from ../src/webgl.h:9,
                 from ../src/bindings.cc:9:
/home/sbrl/.node-gyp/12.2.0/include/node/v8.h:5947:46: note: candidate: ‘v8::MaybeLocal<v8::Function> v8::FunctionTemplate::GetFunction(v8::Local<v8::Context>)’
   V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
                                              ^~~~~~~~~~~
/home/sbrl/.node-gyp/12.2.0/include/node/v8.h:5947:46: note:   candidate expects 1 argument, 0 provided
In file included from ../src/webgl.h:9,
                 from ../src/bindings.cc:9:
../src/bindings.cc: At global scope:
/home/sbrl/.node-gyp/12.2.0/include/node/node.h:556:43: warning: cast between incompatible function types from ‘void (*)(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’ {aka ‘void (*)(v8::Local<v8::Object>)’} to ‘node::addon_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’} [-Wcast-function-type]
       (node::addon_register_func) (regfunc),                          \
                                           ^
/home/sbrl/.node-gyp/12.2.0/include/node/node.h:590:3: note: in expansion of macro ‘NODE_MODULE_X’
   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
   ^~~~~~~~~~~~~
../src/bindings.cc:489:1: note: in expansion of macro ‘NODE_MODULE’
 NODE_MODULE(webgl, Init)
 ^~~~~~~~~~~
In file included from /home/sbrl/.node-gyp/12.2.0/include/node/node.h:63,
                 from ../src/webgl.h:9,
                 from ../src/bindings.cc:9:
/home/sbrl/.node-gyp/12.2.0/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]’:
/home/sbrl/.node-gyp/12.2.0/include/node/node_object_wrap.h:84:78:   required from here
/home/sbrl/.node-gyp/12.2.0/include/node/v8.h:9817:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<node::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
                reinterpret_cast<Callback>(callback), type);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sbrl/.node-gyp/12.2.0/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]’:
../../nan/nan_object_wrap.h:65:61:   required from here
/home/sbrl/.node-gyp/12.2.0/include/node/v8.h:9817:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
make: *** [webgl.target.mk:120: Release/obj.target/webgl/src/bindings.o] Error 1
make: Leaving directory '/tmp/ai/node_modules/gl/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node-v12.2.0-linux-x64/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:196:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:257:12)
gyp ERR! System Linux 4.18.0-20-generic
gyp ERR! command "/usr/local/lib/node-v12.2.0-linux-x64/bin/node" "/usr/local/lib/node-v12.2.0-linux-x64/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /tmp/ai/node_modules/gl
gyp ERR! node -v v12.2.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
npm WARN air-quality-ai@0.0.1 No description
npm WARN air-quality-ai@0.0.1 No repository field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gl@4.2.2 install: `prebuild-install || node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the gl@4.2.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/sbrl/.npm/_logs/2019-05-17T11_47_20_256Z-debug.log
```

 - Full output: https://privatebin.net/?a2d03eea1856e19f#NkTISEj1rnJsxiqj1tb21vPNTj69eRKHPeWMaEEkfTg=
 - npm debug log: [2019-05-17T11_47_20_256Z-debug.log](https://github.com/gpujs/gpu.js/files/3192278/2019-05-17T11_47_20_256Z-debug.log)
 - `uname -a`: `Linux Riikaan 4.18.0-20-generic #21-Ubuntu SMP Mon May 6 18:45:52 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`
 - `npm --version`: `6.9.0`
 - `node --version`: `v12.2.0`

More information available upon request of course.

**Comment by robertleeplummerjr** [comment]
This likely needs to be handled in https://github.com/stackgl/headless-gl, probably opened there.

**Comment by robertleeplummerjr** [comment]
Can you reopen this issue in https://github.com/stackgl/headless-gl?

Going to go ahead and close as it is not actionable here, that I can see.

**Comment by sbrl** [comment]
Sure thing!

**Comment by sbrl** [comment]
[Looks like Node.js v12 support has landed in headless-gl v4.3.0](https://github.com/stackgl/headless-gl/releases/tag/v4.3.0)!

**Comment by sbrl** [comment]
[Looks like Node.js v12 support has landed in headless-gl v4.3.0](https://github.com/stackgl/headless-gl/releases/tag/v4.3.0), @robertleeplummerjr!
